math/binary/sqrt.go

```diff
@@ -0,0 +1,26 @@
+// sqrt.go
+// description: Square root calculation
+// details:
+// Calculating the square root using binary operations and a magic number 0x5f3759df [See more](https://en.wikipedia.org/wiki/Fast_inverse_square_root)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see sqrt_test.go
+
+package binary
+
+import (
+	"math"
+)
+
+const threeHalves = 1.5
+
+func Sqrt(number float32) float64 {
+	var halfHumber, y float32
+	halfHumber = number * 0.5
+	z := math.Float32bits(number) // floating point bit level hacking
+	z = 0x5f3759df - (z >> 1)     // Newton's approximation
+	y = math.Float32frombits(z)
+	for i := 0; i < 3; i++ {
+		y = y * (threeHalves - (halfHumber * y * y))
+	}
+	return math.RoundToEven(float64(1 / y))
```

Comment:
I think this really breaks the deal. The Fast inverse square root alogrithm was created precisely to avoid expensive operations like division. I think you should add it as the inverse square root algorithm only.

Reply:
I don't see how that is relevant?

Reply:
if you do not use rounding, then due to the peculiarities of float, we will get results like 1.00000001 or 0.9999999999 and it seems that this is not very convenient for writing tests and checking values

Reply:
Yes, but I don't how that has anything to do with my comment.

Reply:
> Yes, but I don't how that has anything to do with my comment.

then apparently I did not understand your comment, sorry. I just wanted to give a nice trick of getting a square root without using library functions sqrt

Reply:
The quake III algorithm is a genius thing, so I wanted to add it as is. @tjgurwara99 @siriak What do you guys think?

Reply:
Personally don't think this is bad though. The Quake III game algorithm was built because of the efficiency of calculation of reflections angles that this algorithm provides (if I remember correctly - don't quote me on that though, my physics is bad so... 😂) so if we needed `sqrt` instead of the inverse square root then it makes sense but personally I think this is a valid approach to get an `sqrt` directly.  Also from the benchmarks, it seems to be faster still than the standard library `Sqrt` which in most new architectures amd64, arm64 etc) uses the assembly instructions which should in theory be faster than this so I think it's a good approach.